test/algorithms/is_valid.cpp

```diff
@@ -1433,14 +1433,8 @@ BOOST_AUTO_TEST_CASE( test_with_NaN_coordinates )
     std::cout << "************************************" << std::endl;
 #endif
 
-    linestring_type ls1, ls2;
-    bg::read_wkt("LINESTRING(1 1,1.115235e+308 1.738137e+308)", ls1);
-    bg::read_wkt("LINESTRING(-1 1,1.115235e+308 1.738137e+308)", ls2);
-
-    // the intersection of the two linestrings is a new linestring
-    // (multilinestring with a single element) that has NaN coordinates
     multi_linestring_type mls;
-    bg::intersection(ls1, ls2, mls);
+    bg::read_wkt("MULTILINESTRING((nan nan))", mls);
```

Comment:
Can these changes be explained?

Reply:
I think the last.
I didn't add these cases here, I think @awulkiew  did.
Probably for the reason that the intersection indeed returned something like that, and then the `is_valid` functionality should check these results.

Reply:
It was added by @mkaravel:
https://github.com/boostorg/geometry/commit/73960f31ca3e6483a5846bbf0b6ad601a4640fbe
It seems such geometry was causing infinite loops. I think we should leave this test.